File: 03_Spring/demo/jv_spring/src/main/java/com/springbook/biz/polymorphism/speaker/SonySpeaker.java
package com.springbook.biz.polymorphism.speaker;

public class SonySpeaker implements Speaker {
    public SonySpeaker() {
        System.out.println("===> SonySpeaker 객체 생성");
    }

    public void volumeUp() {
        System.out.println("SonySpeaker Volume Up");
    }

    public void volumeDown() {
        System.out.println("SonySpeaker Volume Down");
    }
}

File: 03_Spring/kotlin_spring/src/main/kotlin/com/example/kotlin_spring/polymorphism/speaker/SonySpeaker.kt
package com.example.kotlin_spring.polymorphism.speaker

class SonySpeaker: Speaker {

    init {
        println("SonySpeaker 객체 생성..")
    }

    override fun volumeUp() {
        println("SonySpeaker Volume Up")
    }

    override fun volumeDown() {
        println("SonySpeaker Volume Down")
    }
}